packages/push/src/push_vm/push_io/mod.rs

```diff
@@ -0,0 +1,15 @@
+use std::{io::Write, string::FromUtf8Error};
+
+pub trait HasStdout {
+    type Stdout: Write;
+
+    fn stdout(&mut self) -> &mut Self::Stdout;
+
+    /// Return the contents of `stdout` as a `String`
+    ///
+    /// # Errors
+    ///
+    /// Returns a `FromUtf8Error` if there is a problem converting
+    /// the contents of `Self::Stdout` into a `String`.
+    fn stdout_string(&mut self) -> Result<String, FromUtf8Error>;
```

Comment:
if this is in this trait I would provide a default impl here
```suggestion
    fn stdout_string(&mut self) -> std::io::Result<String>
    where
        Self::Stdout: Seek
    {
           self.stdout().rewind()?;
           let mut buf = String::new();
           self.stdout().read_to_string(&mut buf)?;
           
           Ok(buf)
    }
```

Reply:
Do we need something this complex that depends on `Seek`? Would it make more sense to just make the default implementation depend on `Cursor` and basically just be the one that we have in `PushState`?

Reply:
We ended up removing `stdout_string` from the trait, so this became a non-issue.